### ISSUE TITLE
Incomplete dataset

### ISSUE THREAD

**Issue by wj7486** [bytedance-fanqie-ai/MOSAIC#8]
Dear author, thank you very much for sharing your code. It's a very meaningful method. However, I encountered some issues during the training process. In this ByteDance-FanQie/SemAlign-MS-Subjects200K dataset, I reviewed all the mask images and found that there are only single-subject masks. During training, the sep_loss is fixed at 0 (this is the training printout [19:24:12<956:29:13, 6.29s/it, align_loss=0.513, diffusion_loss=0.383, loss=0.895, lr=8e-5, sep_loss=-0, timestep=0.754]). I noticed that if the number of reference images is less than two, the loss function value remains fixed at 0. Therefore, I wonder if this dataset is missing multi-subject parts. What should I do next? Looking forward to your reply. Thank you again.

```
def compute_align_sep_losses(attn_map, coords, ref_len):
   ......
    # --- 2. ref-image level decoupling ---
    if num_refs >= 2:
        attn_map_reshaped = attn_map.view(num_refs, ref_len, tgt_len)  # [num_refs, ref_len, tgt_len]
        avg_tokens = attn_map_reshaped.mean(dim=1)  # [num_refs, tgt_len]
        # avg_tokens = F.normalize(avg_tokens, p=2, dim=-1)  # L2 norm
        
        # JSD
        loss_sep = 0.0
        for i in range(num_refs):
            for j in range(i+1, num_refs):
                loss_sep += jsd_loss(avg_tokens[i], avg_tokens[j])
        loss_sep /= (num_refs * (num_refs-1) / 2)

    else:
        loss_sep = torch.tensor(0.0, device=attn_map.device)

    return loss_align, - loss_sep
```

**Comment by sd0809** [comment]
Hi,Good catch! Your understanding of the code is spot on. Since sep_loss is designed to decouple different subjects, it naturally stays at 0 when $num_{refs} < 2$ (single-subject training).

Regarding the data: we can't share the internal multi-reference images due to company data security policies. I've just updated the multi-reference data construction pipeline in the repo. You can use it to process your own images or public datasets into the multi-subject format. 

Hope this clears things up!

**Comment by wj7486** [comment]
> Hi,Good catch! Your understanding of the code is spot on. Since sep_loss is designed to decouple different subjects, it naturally stays at 0 when n u m r e f s < 2 (single-subject training).
> 
> Regarding the data: we can't share the internal multi-reference images due to company data security policies. I've just updated the multi-reference data construction pipeline in the repo. You can use it to process your own images or public datasets into the multi-subject format.
> 
> Hope this clears things up!

Thank you very much for your outstanding contribution. Thank you again.
